File: backend/app/schemas/unanswered.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

UnansweredStatus = Literal["open", "reviewing", "resolved", "dismissed"]


class UnansweredQuestionResponse(BaseModel):
    id: str
    knowledge_base_id: int
    query_text: str
    occurrence_count: int
    first_asked_at: datetime
    last_asked_at: datetime
    status: str
    reason: str
    top_score: float | None
    sample_user_message_id: str | None
    sample_assistant_message_id: str | None
    suggested_title: str | None
    suggested_summary: str | None
    resolved_item_id: int | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class UnansweredQuestionUpdate(BaseModel):
    status: UnansweredStatus | None = None
    suggested_title: str | None = Field(default=None, max_length=200)
    suggested_summary: str | None = Field(default=None, max_length=5000)


class UnansweredResolveRequest(BaseModel):
    title: str | None = Field(default=None, max_length=200)
    content: str | None = Field(default=None, max_length=50000)
    summary: str | None = Field(default=None, max_length=2000)


class UnansweredResolveResponse(BaseModel):
    unanswered: UnansweredQuestionResponse
    knowledge_item_id: int

File: frontend/src/types/unanswered.ts
export type UnansweredStatus = 'open' | 'reviewing' | 'resolved' | 'dismissed'

export type UnansweredReason = 'no_citations' | 'reject_phrase' | 'low_score' | 'no_hits'

export interface UnansweredQuestion {
  id: string
  knowledge_base_id: number
  query_text: string
  occurrence_count: number
  first_asked_at: string
  last_asked_at: string
  status: UnansweredStatus
  reason: UnansweredReason | string
  top_score: number | null
  sample_user_message_id: string | null
  sample_assistant_message_id: string | null
  suggested_title: string | null
  suggested_summary: string | null
  resolved_item_id: number | null
  created_at: string
  updated_at: string
}

export interface UnansweredResolvePayload {
  title?: string
  content?: string
  summary?: string
}

export interface UnansweredResolveResult {
  unanswered: UnansweredQuestion
  knowledge_item_id: number
}

export interface PaginatedUnanswered {
  items: UnansweredQuestion[]
  total: number
  page: number
  page_size: number
}

export const REASON_LABELS: Record<string, string> = {
  no_citations: '无引用',
  reject_phrase: '拒答',
  low_score: '低分检索',
  no_hits: '无检索结果',
}

export const STATUS_LABELS: Record<UnansweredStatus, string> = {
  open: '待处理',
  reviewing: '处理中',
  resolved: '已沉淀',
  dismissed: '已忽略',
}
